compiler/rustc_hir_typeck/src/demand.rs

```diff
@@ -1432,4 +1432,46 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             _ => false,
         }
     }
+
+    pub fn check_for_range_as_method_call(
+        &self,
+        err: &mut Diagnostic,
+        expr: &hir::Expr<'_>,
+        checked_ty: Ty<'tcx>,
+        // FIXME: We should do analysis to see if we can synthesize an expresion that produces
+        // this type for always accurate suggestions, or at least marking the suggestion as
+        // machine applicable.
+        expected_ty: Ty<'tcx>,
+    ) {
+        if !hir::is_range_literal(expr) {
+            return;
+        }
+        let hir::ExprKind::Struct(
+            hir::QPath::LangItem(LangItem::Range, ..),
+            [start, end],
+            _,
+        ) = expr.kind else { return; };
+        let mut expr = end.expr;
+        while let hir::ExprKind::MethodCall(_, rcvr, ..) = expr.kind {
+            // Getting to the root receiver and asserting it is a fn call let's us ignore cases in
+            // `src/test/ui/methods/issues/issue-90315.stderr`.
+            expr = rcvr;
+        }
+        let hir::ExprKind::Call(..) = expr.kind else { return; };
```

Comment:
Ideally we'd check that the lhs of the `..` [has a method](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir_typeck/fn_ctxt/struct.FnCtxt.html#method.probe_for_name) with the name of this call ident. 

You should at least be checking that this call has only one path segment though, since `a..b::c()` is going to be an eroneous suggestion, right?